src/vignette/storage/static_assets.clj

```diff
@@ -0,0 +1,48 @@
+(ns vignette.storage.static-assets
+  (:require [vignette.storage.protocols :refer :all]
+            [org.httpkit.client :as http]
+            [vignette.util.consul :as consul]
+            [vignette.util.filesystem :as fs]
+            [clojure.java.io :as io]))
+
+(defn build-url [oid]
+  (str
+    (consul/->uri
+      (consul/find-service
+        consul/create-consul "static-assets" consul/service-query-tag)) "/image/" oid))
+
+(defn- parse-content-disp
+  [header]
+  (when-let [m (if header (re-find #"(?i)filename=\"(.*)\"" header))]
+    (second m)))
+
+(defrecord AsyncResponseStoredObject [response] StoredObjectProtocol
+  (file-stream [this] (-> this :response :body))
+  (content-length [this] (-> this :response :content-length))
+  (content-type [this] (-> this :response :headers :content-type))
+  (etag [this] (-> this :response :headers :etag))
+  (filename [this] (-> this :response :headers :content-disposition parse-content-disp))
+  (transfer! [this to]
+    (with-open [in-stream (io/input-stream (file-stream this))
+                out-stream (io/output-stream to)]
+      (io/copy in-stream out-stream))
+    (fs/file-exists? to))
+  (->response-object [this] (file-stream this)))
+
+(defn create-async-response-stored-object [oid]
+  (let [response @(http/get (build-url oid) {:as :stream})]
+    (if (-> response :status (= 200))
+      (->AsyncResponseStoredObject response))))
```

Comment:
S3 is proprietary protocol that isn't well standardised to begin with. I've found problems where RadosGW couldn't be used with newer s3 libraries because of encryption scheme change in the upstream S3.

I'm all for reusing technologies but we would have to spend a lot of time to reimplement S3 interface. And the only thing we would gain from this is maybe the possibility of reusing some existing libraries. 

If you guys think that us reimplementing S3 protocol is a must have feature of SAS we should talk about it more. But this is a topic for another discussion. 

Additionally the existing inherited from MW file path scheme and bucket abstraction is something SAS doesn't need to support.